nautobot/ipam/signals.py

```diff
@@ -91,6 +91,20 @@ def ip_address_to_interface_assignment_created(sender, instance, raw=False, **kw
     instance.full_clean()
 
 
+@receiver(m2m_changed, sender=Prefix.locations.through)
+def assert_prefix_locations_content_types(sender, instance, action, reverse, model, pk_set, **kwargs):
+    if action == "pre_add":
+        instance_ct = ContentType.objects.get_for_model(instance)
+        invalid_locations = model.objects.select_related("location_type").filter(
+            Q(pk__in=pk_set), ~Q(location_type__content_types__in=[instance_ct])
+        )
+        if invalid_locations.exists():
+            invalid_location_types = {location.location_type.name for location in invalid_locations}
+            raise ValidationError(
+                {"locations": f"Prefixes may not associate to Locations of types {list(invalid_location_types)}."}
+            )
```

Comment:
Since most of the block of code is duplicate of `nautobot.ipam.signals.assert_vlan_locations_content_types`, would it be better to just move it into a standalone function?

Reply:
I'll see what I can do.

Reply:
Combined the two signals into one. Also found that we were missing handling for the `reverse` case of the signal, i.e. `Location.prefixes.add(prefix)` or `Location.vlans.add(vlan)` so I've added that and test coverage for it.